lib/ClangImporter/ClangImporter.cpp

```diff
@@ -781,6 +799,17 @@ void importer::getNormalInvocationArguments(
     invocationArgStrs.push_back("-iapinotes-modules");
     invocationArgStrs.push_back(path.str().str());
   }
+
+  // If the target triple needs to be replaced, pass the argument at the end of
+  // the invocation.
+  if (updateTargetTriple) {
```

Comment:
I'm not sure if this one required, but I also don't know what is means to get a normal invocation. I notice that  `getNormalInvocationArguments` gets called in both in implicit & explicit modules mode.

Reply:
Couldn't we do this in `addCommonInvocationArguments` instead?

Reply:
I didn't see much helpful output while reproducing issues. I can try some more simple tests out, but i'd prefer to be confident that it's not necessary than the inverse.

Reply:
Probably @cachemeifyoucan knows. I personally hate having multiple `-target` args on the command line...

Reply:
Oh actually, reading the code, Ian is right. There's no code path that goes through `getNormalInvocationArguments` but doesn't go through `addCommonInvocationArguments`

```
ClangImporter::getClangDriverArguments(ASTContext &ctx, bool ignoreClangTarget) {
...
  invocationArgStrs.push_back(ctx.ClangImporterOpts.clangPath);
  switch (ctx.ClangImporterOpts.Mode) {
  case ClangImporterOptions::Modes::Normal:
  case ClangImporterOptions::Modes::PrecompiledModule:
    getNormalInvocationArguments(invocationArgStrs, ctx, ignoreClangTarget);
    break;
...
  }
  addCommonInvocationArguments(invocationArgStrs, ctx,
      requiresBuiltinHeadersInSystemModules, ignoreClangTarget);
  return invocationArgStrs;
}


```